helm/cluster-cloud-director/templates/_ignition.tpl

```diff
@@ -60,6 +60,7 @@ ignition:
             RemainAfterExit=yes
             ExecStart=/usr/bin/bash -cv 'echo "$("$(find /usr/bin /usr/share/oem -name vmtoolsd -type f -executable 2>/dev/null | head -n 1)" --cmd "info-get guestinfo.ignition.network")" > /opt/set-networkd-units'
             {{- if $.Values.connectivity.network.staticRoutes }}
+            ExecStart=/usr/bin/bash -cv 'echo "sleep 3" >> /opt/set-networkd-units'
```

Comment:
Originally I had the routes in a separate unit that had set-networkd-units as a dependency but I had (what I now believe) the same race condition. 

If there's a cleaner way to do this I'll be quite happy with changing it. I did try `while $(systemctl is-active systemd-networkd)` instead of this dumb `sleep` but the command wasn't escaped and I didn't really want to spend time on it.

Reply:
Maybe I could move them back to their own unit and add [this](https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/) to it. Not 100% this would be the fix though, systemd is still a bit obscure to me sometimes.

```
After=network-online.target
Wants=network-online.target
```

Reply:
I did `systemctl stop systemd-networkd`
then `systemctl status network-online.target` and it says `active`
So I guess not then..